File: src/main/java/com/baidu/algorithm/math/_263_Ugly_Number.py
class Solution(object):
    def isUgly(self, num):
        """
        :type num: int
        :rtype: bool
        """
        if num > 0:
            while num % 2 == 0:
                num >>= 1
            while num % 3 == 0:
                num /= 3
            while num % 5 == 0:
                num /= 5

            return True if num == 1 else False

        return False

File: src/main/java/com/baidu/algorithm/math/_263_Ugly_Number.java
/*
 * Copyright (C) 2018 Baidu, Inc. All Rights Reserved.
 */
package com.baidu.algorithm.math;

/**
 * _263_Ugly_Number
 *
 * @author xuhaoran01
 */
public class _263_Ugly_Number {
    public boolean isUgly(int num) {
        if (num < 1) {
            return false;
        }

        int[] factors = new int[]{2, 3, 5};
        for (int factor : factors) {
            while (num % factor == 0) {
                num /= factor;
            }
        }

        return num == 1;
    }
}
